pac4j-http/src/main/java/org/pac4j/http/client/indirect/FormClient.java

```diff
@@ -79,23 +79,36 @@ protected UsernamePasswordCredentials retrieveCredentials(final WebContext conte
             credentials = getCredentialsExtractor().extract(context);
             logger.debug("usernamePasswordCredentials: {}", credentials);
             if (credentials == null) {
-                String redirectionUrl = CommonHelper.addParameter(this.loginUrl, this.usernameParameter, username);
-                redirectionUrl = CommonHelper.addParameter(redirectionUrl, ERROR_PARAMETER, MISSING_FIELD_ERROR);
-                logger.debug("redirectionUrl: {}", redirectionUrl);
-                final String message = "Username and password cannot be blank -> return to the form with error";
-                logger.debug(message);
-                throw RequiresHttpAction.redirect(message, context, redirectionUrl);
+				// it's an AJAX request -> unauthorized (instead of a
+				// redirection)
+				if (getAjaxRequestResolver().isAjax(context)) {
```

Comment:
This is where I realized that both blocks (line 82 - 94 and line 99 - 111) of source code looks very close: can we extract that in a method? Thanks